addons/website_sale/controllers/main.py

```diff
@@ -1408,77 +1408,39 @@ def extra_info(self, **post):
     # Payment
     # ------------------------------------------------------
 
-    def _get_express_shop_payment_values(self, order, **kwargs):
-        logged_in = not request.website.is_public_user()
-        providers_sudo = request.env['payment.provider'].sudo()._get_compatible_providers(
-            order.company_id.id,
-            order.partner_id.id,
-            order.amount_total,
-            currency_id=order.currency_id.id,
-            is_express_checkout=True,
-            sale_order_id=order.id,
-            website_id=request.website.id,
-        )  # In sudo mode to read the fields of providers, order and partner (if not logged in)
-        fees_by_provider = {
-            p_sudo: p_sudo._compute_fees(
-                order.amount_total, order.currency_id, order.partner_id.country_id
-            ) for p_sudo in providers_sudo.filtered('fees_active')
-        }
-        return {
-            # Payment express form values
-            'providers_sudo': providers_sudo,
-            'fees_by_provider': fees_by_provider,
-            'amount': order.amount_total,
+    def _get_express_shop_payment_values(self, order):
+        vals = sale_portal.CustomerPortal._get_payment_values(
+            self, order, website_id=request.website.id, is_express_checkout=True
+        )
+        vals.update({
             'minor_amount': payment_utils.to_minor_currency_units(
-               order.amount_total, order.currency_id
+                order.amount_total, order.currency_id
             ),
             'merchant_name': request.website.name,
-            'currency': order.currency_id,
-            'partner_id': order.partner_id.id if logged_in else -1,
-            'payment_access_token': order._portal_ensure_token(),
             'transaction_route': f'/shop/payment/transaction/{order.id}',
             'express_checkout_route': self._express_checkout_route,
             'landing_route': '/shop/payment/validate',
-        }
+        })
+        vals.pop('tokens')
+        vals.pop('show_tokenize_input')
```

Comment:
I find it ugly as well. The problem is I'm afraid what can be done with this info being rendered. Also adding an argument for this would be ugly as well (bit less but still). Or I could return those values only `if kwargs.get('is_express_checkout')`, but I find it ugly as well to call kwargs from something just passing by.

Reply:
Yes, I understand that, maybe add these two only when an innocent enough looking variable like `include_token_info` is present. :thinking:  This way, no mention of express checkout in a module that don't use it.
Still not entirely convince, since it's a second variable that's only used in 1 flow, but... if we do it for the partner's token, why not for that? We either adapt this method for all the modules, or we change the values after when it's too specific for all.

If we adapt it for all, some comments to explain why these variables are needed could be interesting.

Reply:
I will try with an additionnal argument then, thanks

Reply:
Can't I just replace them here by `request.env['payment.token'] ` and `False`, respectively? 
Best trade-off I ended up with

Reply:
This would be cleaner than poping it, indeed, but it's still sad performance wise.

There is still a double standard here... For `subscription`, you made a change instead of modifying it afterward, but you don't for `website_sale` :thinking:

Reply:
Here you are, I also reordered the steps in the `_get_payment_values`: https://github.com/odoo/odoo/compare/a5108e098568f991aff41dcb65abbe20bf6db670..94c5dbec52d9b0cd28420fa58dd14a68708496e1